lib/reactive_commons/messaging/reply_router.ex

```diff
@@ -2,44 +2,49 @@ defmodule ReplyRouter do
   @moduledoc false
   use GenServer
 
-  @table_name :reply_registry
-
-  def start_link(_) do
-    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  def start_link(broker) do
+    GenServer.start_link(__MODULE__, broker, name: build_name(broker))
   end
 
-  def init(_) do
-    :ets.new(@table_name, [:named_table, read_concurrency: true])
+  @impl true
+  def init(broker) do
+    :ets.new(table_name(broker), [:named_table, read_concurrency: true])
     {:ok, nil}
   end
 
-  def register_reply_route(correlation_id, pid) do
-    GenServer.call(__MODULE__, {:register, correlation_id, pid})
+  def register_reply_route(broker, correlation_id, pid) do
+    GenServer.call(build_name(broker), {:register, broker, correlation_id, pid})
   end
 
-  def delete_reply_route(correlation_id) do
-    GenServer.cast(__MODULE__, {:delete, correlation_id})
+  def delete_reply_route(broker, correlation_id) do
+    GenServer.cast(build_name(broker), {:delete, broker, correlation_id})
   end
 
-  def route_reply(correlation_id, reply_message) do
-    case :ets.lookup(@table_name, correlation_id) do
-      [{^correlation_id, pid}] ->
+  def route_reply(broker, correlation_id, reply_message) do
+    case :ets.lookup(table_name(broker), {to_string(broker), correlation_id}) do
+      [{{broker, ^correlation_id}, pid}] ->
         send(pid, {:reply, correlation_id, reply_message})
-        delete_reply_route(correlation_id)
+        delete_reply_route(broker, correlation_id)
         :ok
 
       [] ->
         :no_route
     end
   end
 
-  def handle_call({:register, correlation_id, pid}, _, _) do
-    :ets.insert(@table_name, {correlation_id, pid})
+  @impl true
+  def handle_call({:register, broker, correlation_id, pid}, _, _) do
+    :ets.insert(table_name(broker), {{to_string(broker), correlation_id}, pid})
     {:reply, :ok, nil}
   end
 
-  def handle_cast({:delete, correlation_id}, _) do
-    :ets.delete(@table_name, correlation_id)
+  @impl true
+  def handle_cast({:delete, broker, correlation_id}, _) do
+    :ets.delete(table_name(broker), {to_string(broker), correlation_id})
     {:noreply, nil}
   end
+
+  defp build_name(broker), do: String.to_atom("reply_router_#{broker}")
+
+  defp table_name(broker), do: String.to_atom("reply_router_table_#{broker}")
```

Comment:
Avoid usage of to_atom, try another strategy or if there is no another strategy use to_existing_atom....